universe/src/server.rs

```diff
@@ -0,0 +1,210 @@
+use std::{
+    net::SocketAddr,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use common::{
+    host_info::{self, HostInfo},
+    region::{self, Region, Zone},
+};
+use pin_project::{pin_project, pinned_drop};
+use proto::universe::universe_client::UniverseClient;
+use proto::universe::universe_server::Universe;
+use proto::universe::{
+    CreateKeyspaceRequest, CreateKeyspaceResponse, ListKeyspacesRequest, ListKeyspacesResponse,
+};
+use tokio::sync::broadcast;
+use tokio_stream::{
+    wrappers::{errors::BroadcastStreamRecvError, BroadcastStream},
+    Stream,
+};
+use tokio_util::sync::CancellationToken;
+use tonic::{Request, Response, Status};
+use tracing::{debug, info, instrument};
+
+use crate::storage::{
+    cassandra::{self, Cassandra},
+    Storage,
+};
+
+/// Implementation of the Universe manager.
+pub struct UniverseServer<S: Storage> {
+    storage: Arc<S>,
+}
+
+impl<S: Storage> UniverseServer<S> {
+    /// Creates a new UniverseServer.
+    pub fn new(storage: Arc<S>) -> Self {
+        Self { storage }
+    }
+}
+
+#[tonic::async_trait]
+impl<S: Storage> Universe for UniverseServer<S> {
+    #[instrument(skip(self))]
+    async fn create_keyspace(
+        &self,
+        request: Request<CreateKeyspaceRequest>,
+    ) -> Result<Response<CreateKeyspaceResponse>, Status> {
+        debug!("Got a create_keyspace request: {:?}", request);
```

Comment:
minor: creating keyspaces is not a super frequent operation, I think this can be info.